src/main/java/org/graylog/integrations/notifications/types/microsoftTeams/TeamsEventNotification.java

```diff
@@ -0,0 +1,201 @@
+                    .addSeverity(Notification.Severity.URGENT)
+                    .addDetail("title", "TeamsEventNotification Failed")
+                    .addDetail("description", errorMessage);
+
+            notificationService.publishIfFirst(systemNotification);
+            throw exp;
+
+        } catch (Exception exp) {
+            throw new EventNotificationException("There was an exception triggering the TeamsEventNotification", exp);
+        }
+    }
+
+    /**
+     * @param ctx
+     * @param config
+     * @return
+     * @throws PermanentEventNotificationException - throws this exception when the custom message template is invalid
+     */
+    TeamsMessage createTeamsMessage(EventNotificationContext ctx, TeamsEventNotificationConfig config) throws PermanentEventNotificationException {
+        //Note: Link names if notify channel or else the channel tag will be plain text.
+        String messageTitle = buildDefaultMessage(ctx);
+        String description = buildMessageDescription(ctx);
+        JsonNode customMessage = null;
+        String template = config.customMessage();
+        if (!isNullOrEmpty(template)) {
+            customMessage = buildCustomMessage(ctx, config, template);
+        }
+
+        return new TeamsMessage(
+                config.color(),
+                config.iconUrl(),
+                messageTitle,
+                customMessage,
+                description
+        );
+    }
+
+    String buildDefaultMessage(EventNotificationContext ctx) {
+        String title = ctx.eventDefinition().map(EventDefinitionDto::title).orElse("Unnamed");
+
+        // Build Message title
+        return String.format(Locale.ROOT, "**Alert %s triggered:**\n", title);
+    }
+
+    private String buildMessageDescription(EventNotificationContext ctx) {
+        String description = ctx.eventDefinition().map(EventDefinitionDto::description).orElse("");
+        return "_" + description + "_";
+    }
+
+
+    JsonNode buildCustomMessage(EventNotificationContext ctx, TeamsEventNotificationConfig config, String template) throws PermanentEventNotificationException {
+        final List<MessageSummary> backlog = getMessageBacklog(ctx, config);
+        Map<String, Object> model = getCustomMessageModel(ctx, config.type(), backlog);
+        try {
+            String facts = templateEngine.transform(template, model);
+            JsonNode factsNode = getMessageDetails(facts);
+            LOG.debug("customMessage: template = {} model = {}", template, model);
+            return factsNode;
+        } catch (Exception e) {
+            String error = "Invalid Custom Message template.";
+            LOG.error(error + "[{}]", e.toString());
+            throw new PermanentEventNotificationException(error + e, e.getCause());
+        }
+    }
+
+    public JsonNode getMessageDetails(String eventFields) {
+        String[] fields = eventFields.split("\\r?\\n");
+        for (String  field: fields){
```

Comment:
Nit: just another space
`for (...){` to `for (...) {`

Reply:
Added missing space